rpm/__init__.py

```diff
@@ -79,21 +94,110 @@ def try_path(path: str) -> bool:
     """
-    Initializes the shim. Tries to load system RPM module and replace itself with it.
+    Helps import rpm in cases where the shared object file extensions include the python
+    interpreter version
+
+    Args:
+        path (Path): the absolute path to the rpm module
+        suffixes (List): a list of extension suffixes valid for the python interpreter
+
+    Returns:
+        None
     """
-    for path in get_system_sitepackages():
-        logger.debug(f"Trying {path}")
+    attempted_modules = []
+    while True:
         try:
-            if try_path(path):
-                logger.debug("Import successfull")
-                return
-        except ShimAlreadyInitializingError:
-            continue
-        except Exception as e:
-            logger.debug(f"Exception: {type(e)}: {e}")
+            importlib.reload(sys.modules[__name__])
+        except ModuleNotFoundError as e:
+            if e.name in attempted_modules:
+                logger.debug(f"Already tried {e.name} in {path}, giving up")
+                raise
+            attempted_modules.append(e.name)
+            logger.debug(
+                f"Module {e.name} not found in {path}, "
+                "looking for any binary extensions with valid suffixes"
+            )
+            try_import_binary_extension(path, e.name, suffixes)
+        else:
+            logger.debug(f"Reloaded {__name__}")
+            return
+
+
+def try_import_binary_extension(path: Path, module: str, suffixes: List) -> bool:
+    """
+    Finds and imports a binary extension in {path} matching {module} and {suffixes}
+
+    Args:
+        path (Path): the path to a module, i.e. /usr/lib64/python3.9/site-packages/rpm/
+        module (str): the name of a module to import, i.e. 'rpm._rpm'
+        suffixes (List[str]): a list of extension suffixes to check for (see
+                              importlib.machinery.EXTENSION_SUFFIXES)
+
+    Returns:
+        True if the module was loaded, False otherwise
+    """
+    # get the child module name, i.e. just '_rpm' from 'rpm._rpm'
+    child = module.rpartition(".")[-1]
+    for suffix in suffixes:
+        # a file named {child}{suffix} may exist in {path}, i.e.
+        #
+        # /usr/lib64/python3.9/site-packages/rpm/_rpm.cpython-39-x86_64-linux-gnu.so
+        #
+        # if so we'll try loading it as rpm._rpm
+        so = path / f"{child}{suffix}"
+        if not so.is_file():
+            logger.debug(f"{so} isn't a file, ignoring")
             continue
+        if load_module_by_path(module, so):
+            return True
+    else:
+        logger.debug(f"No suffixes matched {child} found in {path}, giving up")
+        return False
+
+
+def load_module_by_path(module_name: str, path: Path) -> bool:
+    """
+    Imports a python module by file path
+
+    Args:
+        module_name (str): name of the module to import
+        path (Path): absolute path to the module to import
```

Comment:
```suggestion
        module_name: Name of the module to import.
        path: Absolute path to the module to import.
        
    Returns:
        True if the import succeeded, otherwise False.
```